src/main/java/org/ow2/proactive/procci/request/MixinsService.java

```diff
@@ -0,0 +1,199 @@
+package org.ow2.proactive.procci.request;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.ow2.proactive.procci.model.exception.ClientException;
+import org.ow2.proactive.procci.model.exception.CloudAutomationException;
+import org.ow2.proactive.procci.model.occi.infrastructure.constants.Identifiers;
+import org.ow2.proactive.procci.model.occi.infrastructure.mixin.VMImage;
+import org.ow2.proactive.procci.model.occi.metamodel.Entity;
+import org.ow2.proactive.procci.model.occi.metamodel.Mixin;
+import org.ow2.proactive.procci.model.occi.metamodel.MixinBuilder;
+import org.ow2.proactive.procci.model.occi.metamodel.rendering.MixinRendering;
+import com.google.common.base.Supplier;
+import com.google.common.collect.ImmutableMap;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.type.TypeReference;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by the Activeeon Team on 12/10/16.
+ */
+@Component
+public class MixinsService {
+
+    private final Logger logger = LogManager.getLogger(this);
+
+    private final Map<String, Supplier<MixinBuilder>> providerMixin;
+    @Autowired
+    InstancesService instancesService;
```

Comment:
Why package private visibility? for testing? If yes, please use @VisibleForTesting annotation from Guava.